lib/exgboost/parameters.ex

```diff
@@ -0,0 +1,968 @@
+defmodule EXGBoost.Parameters do
+  @global_params [
+    verbosity: [
+      type: {:custom, EXGBoost.Parameters, :validate_verbosity, []},
+      doc: """
+      Verbosity of printing messages. Valid values are:
+      `:silent`, `:warning`, `:info`, `:debug`.
+      """
+    ],
+    use_rmm: [
+      type: :boolean,
+      default: false,
+      doc: """
+      Whether to use RAPIDS Memory Manager for memory allocation.
+      This This option is only applicable when XGBoost is built (compiled)
+      with the RMM plugin enabled. Valid values are `true` and `false`.
+      """
+    ]
+  ]
+
+  @general_params [
+    booster: [
+      type: {:in, [:gbtree, :gblinear, :dart]},
+      default: :gbtree,
+      doc: """
+      Which booster to use. Valid values are `:gbtree`, `:gblinear`, `:dart`
+          * `:gbtree`: tree-based models
+          * `:gblinear`: linear models
+          * `:dart`: tree-based models with dropouts
+      """
+    ],
+    verbosity: [
+      type: {:custom, EXGBoost.Parameters, :validate_verbosity, []},
+      default: :silent,
+      doc: """
+      Verbosity of printing messages. Valid values are:
+      `:silent`, `:warning`, `:info`, `:debug`
+      """
+    ],
+    validate_parameters: [
+      type: :boolean,
+      default: true,
+      doc: """
+      Whether to perform validation of parameters. If set to `true`, an error
+      will be raised if an invalid parameter is passed to the booster, and
+      EXGBoost will take care of formatting all parameters to the expected
+      input of XGBoost. If set to `false`, the user is responsible for ensuring
+      that all parameters are valid strings which is what XGBoost is expecting.
+      """
+    ],
+    nthread: [
```

Comment:
I think this should also be a configuration value. We have it as a config for EXLA. I can't think of a scenario where you'd want to set this per call to predict/train

Reply:
Changing to 
```elixir
nthread: [
      type: :non_neg_integer,
      default: Application.compile_env(:exgboost, :nthread, 0),
      doc: """
      Number of threads to use for training and prediction. If `0`, then the
      number of threads is set to the number of cores.  This can be set globally
      using the `:exgboost` application environment variable `:nthread`
      or on a per booster basis.  If set globally, the value will be used for
      all boosters unless overridden by a specific booster.

      To set the number of threads globally, add the following to your `config.exs`:

      ````elixir
      config :exgboost, nthread: 4
      ````
      """
    ]
```

Reason is to allow a global config while also allowing per booster config which xgboost supports